### PR TITLE
Nullable types

### PR DESCRIPTION
**Pull request by zapling**
I'm looking into this as a potential lib for use in one of our projects. In our schema we have
```
"type": [
  "string",
  "null"
]
```

It would be nice for these to transfer over to the structs generated so we can do `myStruct.Field == nil`.

@atombender let my know what you think of this

### CODE REVIEW

File: pkg/generator/generate.go [hunk 1]
```diff
@@ -551,7 +553,14 @@ func (g *schemaGenerator) generateType(
 	if len(t.Type) == 0 {
 		return codegen.EmptyInterfaceType{}, nil
 	}
-	if len(t.Type) != 1 {
+	if len(t.Type) == 2 {
```
Comment (atombender) [inline pkg/generator/generate.go:557]: This assumes `null` is always the second element, and that the non-null type is the first one. I would have to support both.

**Review by atombender — CHANGES_REQUESTED** [review]
I haven't thought about how JSON Schema treats nulls, but this looks like the canonical approach. I _think_ this is fine, though it'd be nice to support real unions. But the handling of the order of `null` needs to be fixed.

File: pkg/generator/generate.go (hunk 1 above)
Comment (zapling) [reply pkg/generator/generate.go:557]: Ah right, I added a testcase for this as well

**Review by atombender — APPROVED** [review]
Super!

**Comment by atombender** [comment]
Can you rebase from main to get the updated GitHub action?

**Comment by zapling** [comment]
@atombender done!

**Pull request merged**